src/sentry/utils/snuba.py

```diff
@@ -847,9 +848,31 @@ def raw_query(
     return bulk_raw_query([snuba_params], referrer=referrer, use_cache=use_cache)[0]
 
 
-SnubaQuery = Union[Request, MutableMapping[str, Any]]
 Translator = Callable[[Any], Any]
-RequestQueryBody = tuple[Request, Translator, Translator]
+
+
+@dataclasses.dataclass(frozen=True)
+class SnubaRequest:
+    request: Request
+    referrer: str | None  # this should use the referrer class
```

Comment:
Is this a TODO?

Reply:
Yes, I'll prefix the comment with a TODO once I figure out if we can change this in all the call sites.